main/tests/test_views.py

```diff
@@ -23,31 +22,37 @@ def setUp(self):
 
-    @expectedFailure
     def test_index_view_rendering(self):
         """Testing main page for render needed posts."""
         tz = timezone.get_current_timezone()
         post = Post.objects.create(author=self.user, title='Test', text='superText',
-                                   created_date=datetime(day=1, month=3, year=2016, tzinfo=tz))
+                                   created_date=datetime(day=1, month=3, year=2016, tzinfo=tz),
+                                   published_date=datetime(day=1, month=3, year=2016, tzinfo=tz))
         past_post = Post.objects.create(author=self.user, title='past_est', text='superText',
-                                        created_date=datetime(day=1, month=4, year=2015, tzinfo=tz))
+                                        created_date=datetime(day=1, month=4, year=2015, tzinfo=tz),
+                                        published_date=datetime(day=1, month=4, year=2015, tzinfo=tz))
         future_post = Post.objects.create(author=self.user, title='future_test', text='superText',
-                                          created_date=datetime(day=1, month=4, year=2116, tzinfo=tz))
-        with patch('django.utils.timezone.now', lambda: datetime(day=1, month=4, year=2016, tzinfo=tz)) as test_mock:
-            post.publish()
-            future_post.publish()
-            past_post.publish()
-            test_mock.assert_called_with()
+                                          created_date=datetime(day=1, month=4, year=2116, tzinfo=tz),
+                                          published_date=datetime(day=1, month=4, year=2116, tzinfo=tz))
+        with patch('django.utils.timezone.now', lambda: datetime(day=1, month=1, year=2016, tzinfo=tz)):
+            response = self.client.get(reverse('post_list'))
+            self.assertContains(response, past_post)
+            self.assertNotContains(response, post)
+            self.assertNotContains(response, future_post)
+        with patch('django.utils.timezone.now', lambda: datetime(day=1, month=4, year=2016, tzinfo=tz)):
+            response = self.client.get(reverse('post_list'))
+            self.assertContains(response, past_post)
+            self.assertContains(response, post)
+            self.assertNotContains(response, future_post)
+        with patch('django.utils.timezone.now', lambda: datetime(day=1, month=4, year=3016, tzinfo=tz)):
             response = self.client.get(reverse('post_list'))
-            self.assertContains(response, post.title)
             self.assertContains(response, past_post.title)
+            self.assertContains(response, post)
             self.assertContains(response, future_post)
-
-    def test_index_view(self):
-        """Testing main page."""
         response = self.client.get(reverse('post_list'))
         self.assertEqual(200, response.status_code)
         self.assertTemplateUsed(response, 'main/index.html')
+        self.assertListEqual(list(response.context['posts']), [post, past_post])
```

Comment:
Yes, but also add `assertContains()` checking for smth not related to posts right here.